kotlin/lib/src/main/kotlin/Application.kt

```diff
@@ -0,0 +1,57 @@
+package com.svix.kotlin
+
+import com.svix.kotlin.exceptions.ApiException
+import com.svix.kotlin.internal.apis.ApplicationApi
+import com.svix.kotlin.models.ApplicationIn
+import com.svix.kotlin.models.ApplicationOut
+import com.svix.kotlin.models.ListResponseApplicationOut
+
+class ApplicationListOptions() : ListOptions() {
+    override fun iterator(iterator : String) = apply { super.iterator(iterator) }
+
+    override fun limit(limit : Int) = apply { super.limit(limit) }
+}
+
+class Application internal constructor(debugUrl: String) {
+    val api = ApplicationApi(debugUrl)
+
+    suspend fun list(options: ApplicationListOptions): ListResponseApplicationOut {
+        try {
+            return api.listApplicationsApiV1AppGet(options.iterator, options.limit)
+        } catch (ex: Exception) {
+            throw ApiException(ex)
```

Comment:
I think we should only need to catch internal.ApiExceptions, can these catches be narrowed down? I'd prefer the build to fail if for some reason new exceptions are added in the openapi generated code

Reply:
We have 3 different errors, `ServerException, UnsupportedOperationException, ClientException`, I can explicitly catch them if you prefer.

lmk what you think

Reply:
Do they all inherit from the same basic one? Or are they all completely unrelated? If inherit from the same one, just use that, otherwise please catch all three.

Reply:
They are all RuntimeException, though I don't think it's specific enough

As for catching all of them, I think that kotlin doesn't support multi-catch in the same line, there are 2 options (see https://discuss.kotlinlang.org/t/does-kotlin-have-multi-catch/486/20), I don't like either but leaning towards:
`
try {
    // do some work
} catch (ex: Exception) {
    when(ex) {
        is IllegalAccessException, is IndexOutOfBoundsException -> {
            // handle those above
        }
        else -> throw ex
    }
}
`

Reply:
Bahh, yeah, the when trick looks fine.

Reply:
done. 

We can probably create a util class to do that instead

Reply:
Wrapped it